### ISSUE TITLE
display squashed up on some images

### ISSUE THREAD

**Issue by veehexx** [ventoy/PXE#11]
I've noticed this on a few test images (reproducible with mt86plus_620_64.iso; https://memtest.org/).

iventoy 1.0.12 (occurs on .11 too). default 1024x768.
Proxmox VM default is 1280x800.
^ this is the config used for the below screenshot. I've tried changing resolution to match and other common res's but no solution.

![image](https://github.com/ventoy/PXE/assets/1247183/21ecde2f-1e4a-46be-8b55-b1f8054ab86e)

**Comment by ventoy** [comment]
Run   `cat /ventoy/log`   here and upload the result.

**Comment by veehexx** [comment]
the officially tested `openmediavault_6.0.24-amd64.iso` does the following.

boot menu displays correctly.
![image](https://github.com/ventoy/PXE/assets/1247183/e0aeb46a-c112-4858-8316-1d4a530f8da0)

however, this video is after the iso is booted and i assume is the first page of the installer:
[Screencast_20230717_115541.webm](https://github.com/ventoy/PXE/assets/1247183/dca586a0-90aa-4890-b61c-c71ea4170afd)

**Comment by veehexx** [comment]
tested on real hardware, using memtest86+. dell inspiron 3501 with igpu (intel i3-1115g4)

![IMG_20230716_095014049](https://github.com/ventoy/PXE/assets/1247183/7fa6b83c-81b3-4a1b-bb39-91cae8f65369)

**Comment by Gyppie** [comment]
On debian based distros you have to manually edit grub.cfg file in /boot/grub/grub.cfg on the ISO and change:
"set gfxmode=800x600" to "set gfxmode=auto"
And make sure that "set gfxpayload=keep" is not commented out with a"#" in front.
Then try to boot with different screen resolutions to the one that suits your computer.

**Comment by veehexx** [comment]
@Gyppie - i did find that metioned on certain distrays on my searches before bugging it, so unless i didnt do it right then the issue remains.
also, `mt86plus_620_64.iso` (i know not on the iventoy tested list, but it does boot, plus the issue is present on tested iso's) is not debian based and has neither boot options or a way to change resolution so possible solution there.
i beleive OMV is debian based however. I'll test again soon if i get time/remember.

your info is helpful and good for testing, but also i'm on the viewpoint that this should NOT be required for every use of affected iso's. either need an option to define it as a parameter within iventoy webgui (set it one, always effective on that specific iso) or for it to automatically be done at runtime.

**Comment by veehexx** [comment]
using the same `openmediavault_6.0.24-amd64.iso` file; Left side is iso via CD/DVD boot, Right side is iso via iventoy.

This is the same first installer page; Language selection once the installer option is booted.

![image](https://github.com/ventoy/PXE/assets/1247183/c8393707-fab5-4ec8-af15-5b708f344162)

editing the grub cmdline has no effect.

**Comment by Riches** [comment]
I was having this issue on multiple different images (e.g. Windows and Debian based).

I found that changing the 'Boot Menu Resolution' in 'Boot configuration' in the web admin to '1920x1080' (the same as the native resolution of my display) prevented the issue.

This works for me as all of my target machines have that resolution, but it's obviously not a permanent solution for those that have a fleet with different resolutions.

**Comment by veehexx** [comment]
admittedly i didnt try native client res (tried various upto 1920x1080). But just tried 2560x1440; no fix here via proxmox console.

iventoy menu updates to the desired res, but after booting the chosen iso the proxmox window resizes back to 1280x800 and, afaik, theres nothing i can about that.
i've settled on using 1280x800 for the default iventoy res.

if i manually change the resolution, then the following occurs:
memtest iso resizes back to 1280x800
OMV iso jumps back to its default of 800x600 for the boot menu, and then it automatically switches back to 1280x800 during boot/installation. The previously mentioned grub options for debian, still no fix for me and seem non-functional.

**Comment by azhuge233** [comment]
Changing iVentoy resolution only helps when booting Windows ISOs, doesn't work on Debian 12 or Ubuntu Server.

I tested on one PVE VM, one laptop and two PCs, the working resolution is different for each machine.
- One of my PC has a 3840x2160 monitor, changing iVentoy to 2560x1440 corrects Windows ISO display.
- The other has a 2560x1440 monitor, but I have to change iVentoy to 1024x768 to get it working, which I believe is this PC's BIOS resolution.
- Debian always displays correctly on PVE VMs, but for Windows ISOs I always have to change resolution.

I also tried Ubuntu Desktop/Kali's live system, they display just fine.

> CD/DVD is fine, but iventoy is misbehaving.

True, I tested CD/DVD too (with latest Ventoy), all behaves normally.
#9

**Comment by AnomSanjaya** [comment]
I had same issue in Acer ES1-432, i cant boot any OS (even WinPE & Linux). Change resolution doesnt affected

**Comment by azhuge233** [comment]
> > Changing iVentoy resolution only helps when booting Windows ISOs, doesn't work on Debian 12 or Ubuntu Server.
> > I tested on one PVE VM, one laptop and two PCs, the working resolution is different for each machine.
> > ```
> > * One of my PC has a 3840x2160 monitor, changing iVentoy to 2560x1440 corrects Windows ISO display.
> > 
> > * The other has a 2560x1440 monitor, but I have to change iVentoy to 1024x768 to get it working, which I believe is this PC's BIOS resolution.
> > 
> > * Debian always displays correctly on PVE VMs, but for Windows ISOs I always have to change resolution.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I also tried Ubuntu Desktop/Kali's live system, they display just fine.
> > > CD/DVD is fine, but iventoy is misbehaving.
> > 
> > 
> > True, I tested CD/DVD too (with latest Ventoy), all behaves normally. #9
> 
> I know this is very old, but which resolution did you choose for Windows on PXE? Can't get our Windows PE installer working...

> the working resolution is different for each machine.

Don't remember the exact resolution, I had to try every available iPXE resolution on each machine until it displayed properly.
Not sure about Windows PE, I only tested Windows 10/11 ISO images.

**Comment by JMarcosHP** [comment]
I have the same issue on different isos (Windows 10, 11, server. Debian, OMV, Clonezilla, OpenSuse). On 1080p displays, I need to chance everytime the resolution when I want to boot these isos.

Some pics here (Thinkpad T460p 1080p display, B550-Aorus Elite with RX 7600 and LG 24' IPS 1080p monitor):
![IMG_20240209_231232](https://github.com/ventoy/PXE/assets/58377032/7559f688-fa7b-4f6e-a5d5-718deb798114)
![IMG_20240209_231226](https://github.com/ventoy/PXE/assets/58377032/ab8053c1-a085-4dcc-b5fd-5f6e635db129)
![IMG_20240209_231119](https://github.com/ventoy/PXE/assets/58377032/727aa2ed-5ee3-423d-8d41-ebfa82b86917)
![IMG_20240209_231104](https://github.com/ventoy/PXE/assets/58377032/2e53140e-18f1-47ba-8b8a-f21e7b4fd84b)
![IMG_20240209_230915](https://github.com/ventoy/PXE/assets/58377032/7fc10674-568e-4049-8329-4f38e2f1d632)
![IMG_20240209_224228](https://github.com/ventoy/PXE/assets/58377032/862f6f91-3d30-4a3b-87b6-93eb227168a0)
![IMG_20240209_224349](https://github.com/ventoy/PXE/assets/58377032/cb83cd8f-6c09-4fbb-96ea-46dd7ff05346)

**Comment by maitredede** [comment]
Hello

Like #23 , I have the same unusable screen trying to install `debian-12.5.0-amd64-netinst.iso`. I tries iventoy text or gui mode, many resolutions, debian installer text or gui mode...

Display is remotely accessed through IPMI KVM of my Supermicro server

**Comment by maitredede** [comment]
Also tested the "[supported](debian-11.6.0-amd64-DVD-1.iso)" `debian-11.6.0-amd64-DVD-1.iso`, same behavior...

**Comment by styxadmin** [comment]
Got the same issue, using latest version of 1.0.20

Adjusting the resolution works only sometimes, this should not be something that has to be tinkered with

**Comment by JKL213** [comment]
> On debian based distros you have to manually edit grub.cfg file in /boot/grub/grub.cfg on the ISO and change: "set gfxmode=800x600" to "set gfxmode=auto" And make sure that "set gfxpayload=keep" is not commented out with a"#" in front. Then try to boot with different screen resolutions to the one that suits your computer.

As mentioned here, try to set the GfxMode to Auto. This worked in our network. But it's a bit tedious to do on every single ISO.

**Comment by jjyoo38** [comment]
Many people have been suffering from this problem for a long time. When will this bug be resolved??

**Comment by wociscz** [comment]
Finally found that it is an iventoy issue, after hours trying to fix this on the Proxmox side.

**Comment by kort3x** [comment]
Shame this has been an issue for this long.
We were thinking about licensing iVentoy at multiple sites but this is a showstopper.

**Comment by jjyoo38** [comment]
I configured the iPXE server from the link https://rpi4cluster.com/pxe-setup/ 
and it works perfectly. iventoy is no longer needed. 
From the left menu, proceed in this order: Setup -> iPXE -> Windows 10. 
When editing Samba's smb.conf, add 'guest ok = yes'
If you don't do that, permission problems will occur. It's a bit more difficult to build than iventoy, but it works perfectly.

**Comment by binhex** [comment]
waiting on a fix for this as it was the first issue i hit once i had iventoy setup, hit this with gparted in UEFI mode, switched to BIOS and it did fix it so have a work around for that particular tool.

**Comment by smileymattj** [comment]
@jjyoo38 

You have a typo on your blog

`make bin-x86_64-efi/ipxe.efi EMBED=embend.ipxe`

should be:

`make bin-x86_64-efi/ipxe.efi EMBED=embed.ipxe`

an "n" snuck in there

**Comment by smileymattj** [comment]
@jjyoo38 

> 1. When editing Samba's smb.conf, add 'guest ok = yes' to [pxe-boot] section
>    and 'client max protocol = SMB3' to [global] section ( it makes faster loading speed on first samba connect)


guest ok is needed.  

But “client max protocol = SMB3” will have no affect.   I believe you experienced some caching that resulted in faster speeds after applying this setting and doing the transfer a second go around.  

Samba can operate as a server or a client.  (Host a share, or connect to a share) As a PXE server, It’s being used in server configuration.   So client settings wouldn’t apply to the PXE server use.   server max protocol would be the one to make a difference.  

The current samba default is 3_11, so no change should be needed.  
https://www.samba.org/samba/docs/4.9/man-html/smb.conf.5.html#SERVERMAXPROTOCOL

**Comment by jjyoo38** [comment]
> @jjyoo38
> 
> You have a typo on your blog
> 
> `make bin-x86_64-efi/ipxe.efi EMBED=embend.ipxe`
> 
> should be:
> 
> `make bin-x86_64-efi/ipxe.efi EMBED=embed.ipxe`
> 
> an "n" snuck in there
> 
> You've also got a misunderstanding. You don't need to make changes to add NTFS & ping. If you look at the code, it's **NFS** is not NTFS. You're adding NFS support. Which isn't needed for Windows. I didn't see where NFS or ping was used or needed in your tutorial. Also NTFS is not needed till you get loaded into boot.wim, in-which you're now in a windows environment, so NTFS support is implied.
> 
> ```
> cd ~/ipxe/src
> 
> # NTFS support
> sed -i 's/#undef\tDOWNLOAD_PROTO_NFS/#define\tDOWNLOAD_PROTO_NFS/' config/general.h
> 
> # Ping support
> sed -i 's/\/\/#define\ PING_CMD/#define\ PING_CMD/' config/general.h
> sed -i 's/\/\/#define\ IPSTAT_CMD/#define\ IPSTAT_CMD/' config/general.h
> sed -i 's/\/\/#define\ REBOOT_CMD/#define\ REBOOT_CMD/' config/general.h
> sed -i 's/\/\/#define\ POWEROFF/#define\ POWEROFF/' config/general.
> ```
> 
> You can skip the whole compiling process by making putting _embed.ipxe_ in the tftp directory as _autoexec.ipxe_ instead. And using the precompiled binaries provided by ipxe.org.

this is not my blog .....

**Comment by Vyerni** [comment]
Disappointing this is still an issue. 

Makes life a lot harder

**Comment by peddy22** [comment]
pretty much unusable for all my images (kubuntu, windows 11, hirens) - what do people actually use iventoy for?

**Comment by Vyerni** [comment]
> pretty much unusable for all my images (kubuntu, windows 11, hirens) - what do people actually use iventoy for?

At this point I'm starting to reconsider trying to get Netbootxyz going with Local assets. But it's a bit above my knowledge. 

iVentoy is quite difficult I've found.

**Comment by popy2k14** [comment]
Same issue here for memtest86+ v7.20 and urbackup restore stick v2.5.2.

Is there an solution?
Anyone of you have a working memtest?

**Comment by omiinaya** [comment]
yeah, i get this on every image i try to boot.

**Comment by xn1337** [comment]
I tried everything, then swapped out the HDMI Monitor on the pc's i was building (Ryzen 5 8600G) for a 1920x1080 Display-Port version.
Fixed the issue.
on an msi laptop, it took changing the resolution to 1920x1080

**Comment by zircuitz** [comment]
Can confirm this issue is also happening on Windows 11 LTSC setup. Display is completely mangled. Doesn't matter what resolution I set.

**Comment by gaeldb** [comment]
Same problem..
Changing resolution does not change the problem in my case
Best way was to use only live ISO and install from the live OS

**Comment by blitz2kx** [comment]
Same problem here. Clonezilla unusable. Was so happy to get iventoy up and running but this is quite a buzzkill.

**Comment by Tylerjet** [comment]
Issue still very much present has anyone found a workaround as resolution changes seem to work 25% of the time, Or should I start looking at alternatives instead.

**Comment by Walkers-Lab** [comment]
Is anyone aware of a solution for this? I'm trying to PXE boot a Windows 11 ISO but am getting weird lines on the screen. I've tried multiple devices and resolutions but still no further :/

**Comment by smileymattj** [comment]
@Walkers-Lab 
I resolved it by just doing a Vanilla iPXE server.  That's what VentoyPXE is built off of.  Kinda like building a "LAMP" server.  You got to install several parts and configure them work together.  

The things you'll need are:

- iPXE (ipxe.org)
- tftp server (tftpd-hpa)
- http server (nginx, apache, etc...)
- smb server (samba)
- wimboot (github.com/ipxe/wimboot)

If your current network's DHCP server doesn't support pointing to a tftp server boot file, you may want to install one that can, and have your iPXE server do DHCP as well.

**Comment by peddy22** [comment]
@Walkers-Lab 

we just do an autounattend.xml in the ISO to automate the install, although the installation process is not visible, it loads to the desktop fine after installation completes.

**Comment by popy2k14** [comment]
@ventoy an news on this. This project is such a nice "toy" for our home labs and working exceptional well, but yeah, sadly there is this bug.

**Comment by bfenwick8798** [comment]
Still happens with archiso :/
Would be nice to have a fix since I don't have a usb stick but i will try other methods

**Comment by tatay-dyeyd** [comment]
> Finally found that it is an iventoy issue, after hours trying to fix this on the Proxmox side. Output is broken when using UEFI boot. BIOS boot is not affected.

how to boot into BIOS boot?
